AlphaWallet/Tokens/Logic/GetContractInteractions.swift

```diff
@@ -37,11 +31,11 @@ extension GetContractInteractions {
 extension GetContractInteractions.functional {
 
     //TODO rename this since it might include ERC721 (blockscout and compatible like Polygon's). Or can we make this really fetch ERC20, maybe by filtering the results?
-    static func getErc20Interactions(walletAddress: AlphaWallet.Address, server: RPCServer, startBlock: Int? = nil, queue: DispatchQueue) -> Promise<[TransactionInstance]> {
+    static func getErc20Interactions(walletAddress: AlphaWallet.Address, server: RPCServer, startBlock: Int? = nil) -> Promise<[TransactionInstance]> {
         guard let etherscanURL = server.getEtherscanURLForTokenTransactionHistory(for: walletAddress, startBlock: startBlock) else { return .value([]) }
         return firstly {
-            Alamofire.request(etherscanURL).validate().responseJSON(queue: queue, options: [])
-        }.map(on: queue) { rawJson, _ in
+            Alamofire.request(etherscanURL).validate().responseJSON(queue: .global(), options: [])
```

Comment:
The `queue` arg for `responseJSON(queue:options:)` is just the queue which Alamofire would send the response after processing right? i.e. the actual waiting for the HTTP response, parsing, converting to JSON, is done in its own queue/thread, we don't control that? If we leave it out, it falls back to the main queue and then the `map() {}` in the next line below will just run it on the argument we pass to `map(on:)`.

Then we can leave out the `.global()` completely so it's just `responseJSON(options: [])`.

The `map(on:)` in the next line, we generally can/should remove the queue and just do `map() {}` which defaults to main for PromiseKit. But in this particular case (I remember this can be a slow operation with many transaction records, so it' some of those exceptions), I think we have to pass in `.global()` so it looks like this:

```
Alamofire.request(etherscanURL).validate().responseJSON(options: [])
}.map(on: .global()) { rawJson, _ in
```

Or alternatively, a separate non-main queue instead of `.global()`. Maybe it makes sense to create one for such processing, so it's easier to find, manage and observe too?

Does that sound right?

Reply:
Similarly for the other few changes in this commit (I haven't checked the other commit in the PR as of writing this)